binutils: Apply upstream fix to no longer depend on texinfo

bernhardkaindl opened the pull request:
Upstream fixed the bug where binutils started to call makeinfo inadvertently.

Apply the upstream fix and drop the temporary dependency on texinfo.

haampie commented:
Did you try this? It doesn't work for latest binutils, it references files that don't exist for old binutils.

bkaindl committed:
binutils: Apply upstream fix to no longer depend on texinfo

Upstream fixed the bug where binutils started to call makeinfo inadvertently.
Apply the upstream fix and drop the temporary dependency on texinfo.

bernhardkaindl reviewed (commented):
I've checked that the patches only change those places which need fixing,
and tested it with `for i in binutils@2.{33,34,35,36,37,38,39,40};do bin/spack install $i|| break;done`

alalazo commented:
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this. Also, we now have more recent versions of binutils where this annoying bug is fixed.
